docs/reference/api/docker_remote_api_v1.24.md

```diff
@@ -461,6 +462,20 @@ Json Parameters:
     -   **CgroupParent** - Path to `cgroups` under which the container's `cgroup` is created. If the path is not absolute, the path is considered to be relative to the `cgroups` path of the init process. Cgroups are created if they do not already exist.
     -   **VolumeDriver** - Driver that this container users to mount volumes.
     -   **ShmSize** - Size of `/dev/shm` in bytes. The size must be greater than 0.  If omitted the system uses 64MB.
+    -   **Mounts** - List of mount configurations to create the container with. This replaces `Binds` and `Volumes`. If `Mounts` is specified along with either `Binds` or `Volumes`, the request will be rejected. Fields for a mount configuration:
+          + **Type** - String specifying the type of mount, e.g. `hostbind`, `ephemeral`, or `persistent` (required)
+          + **Source** - String specifying the host path to use for the mount. Path must exist. (required, `hostbind` only)
+          + **Destination** - String specifying the container destination path for the mount (required)
+          + **Driver** - String specifying the driver to use when creating a volume (required, `persistent` only)
+          + **Name** - String specifying the name of the volume to create (required, `persistent` only)
+          + **Mode** - comma sepparated list of mount modes, e.g. `ro`, `rw`, `nocopy`
+          + **CreateOpts** - A map of key/value pairs (strings) to use when creating the volume, passed directly to the volume driver. (`persistent only)
```

Comment:
Made I should take `CreateOpts` and `Driver` out from this API and just never create (except for the `ephemeral` case).
This is also inline with how I am not creating non-existing host paths for `hostbind`s.

(note that today `docker run -v foo:/bar`  will do a create if `foo` doesn't exist)